src/parser.rs

```diff
@@ -283,6 +284,31 @@ impl<'a> Parser<'a> {
                 },
             }, // End of Token::Word
             Token::Mult => Ok(Expr::Wildcard),
+            Token::Tilde => Ok(Expr::UnaryOp {
+                op: UnaryOperator::PGBitwiseNot,
+                expr: Box::new(self.parse_subexpr(0)?),
+                infix: false,
+            }),
+            Token::DoubleExclamationMark => Ok(Expr::UnaryOp {
+                op: UnaryOperator::PGInfixFactorial,
+                expr: Box::new(self.parse_subexpr(0)?),
+                infix: false,
+            }),
+            Token::SquareRoot => Ok(Expr::UnaryOp {
+                op: UnaryOperator::PGSqrt,
+                expr: Box::new(self.parse_subexpr(0)?),
+                infix: false,
+            }),
+            Token::CubeRoot => Ok(Expr::UnaryOp {
+                op: UnaryOperator::PGCbrt,
+                expr: Box::new(self.parse_subexpr(0)?),
```

Comment:
This is probably not right, as this parses `WHERE ||/27 = 3` as `||/ ( 27=3 )`, which is not a valid expression. I don't see precedence mentioned in the documentation you linked to, and I think a better guess would be to default all the unary operators to the `PLUS_MINUS_PREC` (moving the new ops to the `tok @` case below).